internal/core/server/workspace.go

```diff
@@ -704,8 +705,71 @@ func (ws *Workspace) DescribeProcesses(ctx context.Context, input *api.DescribeP
 				ID:       component.ID,
 				Name:     component.Name,
 				Provider: "docker",
-				Running:  state.Running,
 			}
+
+			containerInfo, err := ws.Docker.ContainerInspect(ctx, state.ContainerID)
+			if err == nil {
+				process.Running = containerInfo.State.Running
+				if process.Running {
+					statRequest, err := ws.Docker.ContainerStats(ctx, state.ContainerID, false)
+					if err != nil {
+						return nil, fmt.Errorf("could not get container stats: %w", err)
+					}
+					defer statRequest.Body.Close()
+
+					scanner := bufio.NewScanner(statRequest.Body)
+					didScan := scanner.Scan()
+					if err = scanner.Err(); err != nil {
+						return nil, fmt.Errorf("error scanning container stats: %w", err)
+					}
+
+					if !didScan {
+						return nil, fmt.Errorf("could not read container stats")
+					}
+
+					var containerStats DockerStats
+					err = json.Unmarshal(scanner.Bytes(), &containerStats)
+					if err != nil {
```

Comment:
nit: when possible, reduce scope of newly introduced err variable with `if err := json.Unmarshal(...); err != nil {` When I see `err =` without `:`, I go looking elsewhere to see how these errors are handled.